### PR TITLE
Minimal Inplace accumulation testing

### PR DESCRIPTION
**Pull request by oxinabox**
This is the much more minimal version of  #57  to just test Inplace accumulation.

This is waiting on me building and metatesting tools to catch the negative cases

This is currently built on top of https://github.com/JuliaDiff/ChainRulesTestUtils.jl/pull/60/files
I will rebase after that is merged

### CODE REVIEW

File: test/testers.jl
Comment (oxinabox) [inline test/testers.jl:50]: Its hard to find forward-mode rules that are both simple and can be done in-place.
(first on a array of arrays is one of them i guess)

But i don't think we actually need something inplace to test this.

The real test of this functionality in detail is in the test for `check_result.jl`

File: test/testers.jl
Comment (oxinabox) [inline test/testers.jl:75]: `bad_first` isn't bad, but the sensitivities we define for it are.
should i name it something else?

File: test/testers.jl
Comment (oxinabox) [inline test/testers.jl:100]: I manually inspected the output of these to make sure htey were indeed failing in the way we expected

File: test/testers.jl
Comment (nickrobinson251) [reply test/testers.jl:75]: just `_first`? or `my_first`?

**Review by nickrobinson251 — APPROVED** [review]
seems right to me

**Comment by oxinabox** [comment]
I will validate that the Integration test failures are not false positives before merging this

**Comment by oxinabox** [comment]
Will merge this once CI passes

**Pull request merged**
